File: segundos 4 lenguajes/Java/JugadorBase.java
package Java;
abstract class JugadorBase {
    String nombre;
    int nivel;

    public JugadorBase() {
        nombre = "Juan";
        nivel = 10;
    }

    public JugadorBase(String nombre, int nivel) {
        this.nombre = nombre;
        this.nivel = nivel;
    }

    public JugadorBase(String nombre) {
        this.nombre = nombre;
        this.nivel = 1;
    }

    public JugadorBase(JugadorBase j) {
        this.nombre = j.nombre;
        this.nivel = j.nivel;
    }

    abstract void atacar();
}

File: segundos 4 lenguajes/go/JugadorBase.go
package main

type JugadorBase struct {
	Nombre string
	Nivel  int
}

func NewJugadorBase() JugadorBase {
	return JugadorBase{"Juan", 10}
}

func NewJugadorBaseParams(nombre string, nivel int) JugadorBase {
	return JugadorBase{nombre, nivel}
}

func NewJugadorBaseNombre(nombre string) JugadorBase {
	return JugadorBase{nombre, 1}
}

func NewJugadorBaseCopy(j JugadorBase) JugadorBase {
	return JugadorBase{j.Nombre, j.Nivel}
}

// Método abstracto simulado con interfaz
type AbstractBase interface {
	Atacar()
}
